cms/djangoapps/contentstore/management/commands/set_advanced_fields_all_courses.py

```diff
@@ -0,0 +1,248 @@
+            course_ids = [x.id for x in courses]
+
+            failed_update_advanced_fields_courses = []
+
+            for course_id in course_ids[:limit_bulk_update_operation]:  # pylint: disable=redefined-argument-from-local
+                try:
+                    course_module = modulestore().get_course(CourseKey.from_string(str(course_id)))
+
+                    revised_update_fields = self._force_catalog_visibility_about(
+                        update_fields, course_module
+                    )
+
+                    print(f"Applying revised_update_fields = {revised_update_fields} to course {course_id} using `{user}` user.\n")  # pylint: disable=line-too-long
+                    updated_data = update_course_advanced_settings(
+                        course_module, revised_update_fields, user
+                    )
+
+                    print("=" * 80)
+
+                    for field in revised_update_fields.keys():
+                        print(f"Updated data for course advanced settings `{field}` = {updated_data[field]}")  # pylint: disable=line-too-long
+                        print("=" * 80)
+
+                except Exception as err:  # pylint: disable=broad-except
+                    failed_update_advanced_fields_courses.append(str(course_id))
+                    print("=" * 30 + f"> Oops, failed to update fields for {course_id}")
+                    print("Error:")
+                    print(err)
+
+            print("=" * 80)
+            print("=" * 30 + "> Reset fields summary")
+            print(f"Total number of courses needed to update advanced fields: {len(courses)}")
+            print(f"Limited the bulk update operation to {limit_bulk_update_operation} courses to avoid using all memory on server.")  # pylint: disable=line-too-long
+            print(f"Total number of courses which failed to update advanced fields: {len(failed_update_advanced_fields_courses)}")  # pylint: disable=line-too-long
+            print("List of updated advanced fields failed courses ids:")
+            print("\n".join(failed_update_advanced_fields_courses))
+
+            if self._force_update_catalog_visibility_about:
+                print("List of updated courses ids `catalog_visibility` changes from `none` to `about`:")  # pylint: disable=line-too-long
+                print("\n".join(self._updated_catalog_visilibity_about))
+
+            print("=" * 80)
+
+    def _force_catalog_visibility_about(self, update_fields, course_module):
+        """
+        Explicitly update `catalog_visilibility` = `none` to `about` to fix issue with
+        `There was an error loading this course`.
+        Details: https://discuss.openedx.org/t/how-to-programmatically-set-a-course-advanced-fields/8901/3  # pylint: disable=line-too-long
+
+        Return fields to update with the following settings added.
+
+        {
+            "catalog_visibility": {
+                "value": "about"
+            }
+        }
+        """
```

Comment:
If you are interested, I think there is a way to make this new command much more generic and remove the specific `force-update-catalog-visibility-about` flag.

You could instead add an `if-fields-json` flag like this:

```
./manage.py cms set_advanced_fields_all_courses \
    --if-fields-json '{"catalog_visibility": "none"}' \
    --fields-json '{"catalog_visibility": "about"}' \
    --user edx@example.com
```

That means "find all courses that have `catalog_visibility: none` and change them to `catalog_visibility: about`".

The nice thing is it's much more generic and flexible this way, _and_ even if this can only be used for ~100 courses at a time, you can just re-run the exact same command as many times as needed until all courses are updated, because it will only "count" courses that match the `if-fields-json` filter.